File: app/src/main/java/com/veganbeauty/app/data/repository/CartRepository.java
package com.veganbeauty.app.data.repository;

import com.veganbeauty.app.data.local.dao.CartDao;
import com.veganbeauty.app.data.local.entities.CartItemEntity;

import java.util.List;
import kotlinx.coroutines.flow.Flow;

public class CartRepository {
    private final CartDao cartDao;
    private final Flow<List<CartItemEntity>> allCartItems;

    public CartRepository(CartDao cartDao) {
        this.cartDao = cartDao;
        this.allCartItems = cartDao.getAllCartItems();
    }

    public Flow<List<CartItemEntity>> getAllCartItems() {
        return allCartItems;
    }

    public void insertOrUpdate(CartItemEntity item) {
        cartDao.insertCartItem(item);
    }

    public void update(CartItemEntity item) {
        cartDao.updateCartItem(item);
    }

    public void delete(CartItemEntity item) {
        cartDao.deleteCartItem(item);
    }

    public void clear() {
        cartDao.clearCart();
    }
}

File: app/src/main/java/com/veganbeauty/app/data/repository/CartRepository.kt
package com.veganbeauty.app.data.repository

import com.veganbeauty.app.data.local.dao.CartDao
import com.veganbeauty.app.data.local.entities.CartItemEntity
import kotlinx.coroutines.flow.Flow

class CartRepository(private val cartDao: CartDao) {
    val allCartItems: Flow<List<CartItemEntity>> = cartDao.getAllCartItems()

    suspend fun insertOrUpdate(item: CartItemEntity) {
        cartDao.insertCartItem(item)
    }

    suspend fun update(item: CartItemEntity) {
        cartDao.updateCartItem(item)
    }

    suspend fun delete(item: CartItemEntity) {
        cartDao.deleteCartItem(item)
    }

    suspend fun clear() {
        cartDao.clearCart()
    }
}
